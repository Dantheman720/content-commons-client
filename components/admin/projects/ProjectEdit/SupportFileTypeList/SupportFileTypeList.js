/**
 *
 * SupportFileTypeList
 *
 */

import React, { Fragment } from 'react';
import {
  array, bool, object, string
} from 'prop-types';
import { Button } from 'semantic-ui-react';

import EditSupportFiles from 'components/admin/projects/ProjectEdit/EditSupportFiles/EditSupportFiles';
import EditSupportFilesContent from 'components/admin/projects/ProjectEdit/EditSupportFilesContent/EditSupportFilesContent';
import IconPopup from 'components/admin/projects/ProjectEdit/IconPopup/IconPopup';
import SupportItem from 'components/admin/projects/ProjectEdit/SupportItem/SupportItem';
import Placeholder from 'components/admin/projects/shared/Placeholder/Placeholder';

/* eslint-disable react/prefer-stateless-function */
class SupportFileTypeList extends React.PureComponent {
  state = {}

  toggleEditModal = () => (
    this.setState( prevState => (
      { isEditing: !prevState.isEditing }
    ) )
  )

  renderSupportItem = item => {
    const {
      projectId,
      fileType,
      hasSubmittedData
    } = this.props;

    if ( hasSubmittedData ) {
      return (
        <SupportItem
          key={ `${fileType}-${item.id}` }
          projectId={ { ...projectId } }
          fileType={ fileType }
          itemId={ item.id }
        />
      );
    }

    return (
      <Placeholder
        key={ `${fileType}-${item.id}` }
        parentEl="li"
        childEl="span"
        parentStyles={ {
          display: 'flex',
          justifyContent: 'space-between'
        } }
        childStyles={ {
          fileName: { width: '75%' },
          language: {
            width: '20%',
            marginRight: '0',
            backgroundColor: '#5b616b'
          }
        } }
      />
    );
  }

  render() {
    const {
      headline,
      fileType,
      popupMsg,
      data,
      hasSubmittedData,
      hasUploaded
    } = this.props;

    if ( !data.length ) return null;

    return (
      <Fragment>
        <h3>{ `${headline} ` }
          { hasSubmittedData
            && (
              <Fragment>
                <IconPopup
                  message={ popupMsg }
                  iconSize="small"
                  iconType="info circle"
                  popupSize="mini"
                />
                { hasUploaded
                  && (
                    <EditSupportFiles
                      triggerProps={ {
                        className: 'btn--edit',
                        content: 'Edit',
                        size: 'small',
                        basic: true,
                        onClick: this.toggleEditModal
                      } }
                      contentProps={ {
                        data,
                        fileType,
                        closeEditModal: this.toggleEditModal
                      } }
                      modalTrigger={ Button }
                      modalContent={ EditSupportFilesContent }
                      options={ {
                        closeIcon: true,
                        onClose: this.toggleEditModal,
                        open: this.state.isEditing
                      } }
                    />
                  ) }
              </Fragment>
            ) }
        </h3>
        <ul>
          { data.map( this.renderSupportItem ) }
        </ul>
      </Fragment>
    );
  }
}

SupportFileTypeList.propTypes = {
  headline: string,
  projectId: object.isRequired,
  fileType: string,
  popupMsg: string,
  data: array.isRequired,
  hasSubmittedData: bool,
  hasUploaded: bool
};

export default SupportFileTypeList;
