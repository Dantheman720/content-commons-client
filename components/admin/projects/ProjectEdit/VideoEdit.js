import React from 'react';

const VideoEdit = () => (
  <div>Video Edit</div>
);

export default VideoEdit;
