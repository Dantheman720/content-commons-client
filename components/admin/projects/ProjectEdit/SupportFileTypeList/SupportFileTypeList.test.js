import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { projects, supportFilesConfig } from 'components/admin/projects/ProjectEdit/mockData';
import SupportFileTypeList from './SupportFileTypeList';

const props = {
  headline: supportFilesConfig.srt.headline,
  projectId: { videoID: 'made-in-america' },
  fileType: supportFilesConfig.srt.fileType,
  popupMsg: supportFilesConfig.srt.popupMsg,
  data: projects[0].supportFiles.srt,
  hasSubmittedData: false,
  hasUploaded: false
};

const Component = <SupportFileTypeList { ...props } />;

describe( '<SupportFileTypeList />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );
    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders initially `Placeholder` components', () => {
    const wrapper = shallow( Component );
    const placeholders = wrapper.find( 'Placeholder' );
    expect( placeholders ).toHaveLength( props.data.length );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders `SupportItem` components if `hasSubmittedData`', () => {
    const wrapper = shallow( Component );

    expect( wrapper.find( 'SupportItem' ) )
      .toHaveLength( 0 );
    wrapper.setProps( { hasSubmittedData: true } );
    expect( wrapper.find( 'SupportItem' ) )
      .toHaveLength( props.data.length );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders `IconPopup` components if `hasSubmittedData`', () => {
    const wrapper = shallow( Component );

    expect( wrapper.find( 'IconPopup' ) )
      .toHaveLength( 0 );
    wrapper.setProps( { hasSubmittedData: true } );
    expect( wrapper.find( 'IconPopup' ).exists() )
      .toEqual( true );
    expect( wrapper.find( 'IconPopup' ).prop( 'message' ) )
      .toEqual( props.popupMsg );
  } );

  it( 'renders `EditSupportFiles` components if `hasUploaded`', () => {
    const wrapper = shallow( Component );

    expect( wrapper.find( 'EditSupportFiles' ) )
      .toHaveLength( 0 );
    wrapper.setProps( {
      hasSubmittedData: true,
      hasUploaded: true
    } );
    expect( wrapper.find( 'EditSupportFiles' ).exists() )
      .toEqual( true );
  } );

  it( '`SupportItem` components receive correct `projectId` and `itemId`', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { hasSubmittedData: true } );
    const items = wrapper.find( 'SupportItem' );

    expect( items ).toHaveLength( props.data.length );
    items.forEach( ( item, i ) => {
      expect( item.prop( 'projectId' ) )
        .toEqual( props.projectId );
      expect( item.prop( 'itemId' ) )
        .toEqual( props.data[i].id );
    } );
  } );

  it( 'returns null if an empty data array is received', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { data: [] } );
    expect( wrapper.html() ).toEqual( null );
  } );
} );
