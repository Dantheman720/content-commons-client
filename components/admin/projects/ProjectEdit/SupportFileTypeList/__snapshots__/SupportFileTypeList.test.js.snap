// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SupportFileTypeList /> renders \`SupportItem\` components if \`hasSubmittedData\` 1`] = `
<Fragment>
  <h3>
    SRT Files 
    <IconPopup
      iconSize="small"
      iconType="info circle"
      message="Some info about what SRT files are."
      popupSize="mini"
    />
  </h3>
  <ul>
    <SupportItem
      fileType="srt"
      itemId="5678"
      key="srt-5678"
      projectId={
        Object {
          "videoID": "made-in-america",
        }
      }
    />
    <SupportItem
      fileType="srt"
      itemId="5679"
      key="srt-5679"
      projectId={
        Object {
          "videoID": "made-in-america",
        }
      }
    />
    <SupportItem
      fileType="srt"
      itemId="5680"
      key="srt-5680"
      projectId={
        Object {
          "videoID": "made-in-america",
        }
      }
    />
    <SupportItem
      fileType="srt"
      itemId="5681"
      key="srt-5681"
      projectId={
        Object {
          "videoID": "made-in-america",
        }
      }
    />
  </ul>
</Fragment>
`;

exports[`<SupportFileTypeList /> renders initially \`Placeholder\` components 1`] = `
<Fragment>
  <h3>
    SRT Files 
  </h3>
  <ul>
    <Placeholder
      childEl="span"
      childStyles={
        Object {
          "fileName": Object {
            "width": "75%",
          },
          "language": Object {
            "backgroundColor": "#5b616b",
            "marginRight": "0",
            "width": "20%",
          },
        }
      }
      key="srt-5678"
      parentEl="li"
      parentStyles={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    />
    <Placeholder
      childEl="span"
      childStyles={
        Object {
          "fileName": Object {
            "width": "75%",
          },
          "language": Object {
            "backgroundColor": "#5b616b",
            "marginRight": "0",
            "width": "20%",
          },
        }
      }
      key="srt-5679"
      parentEl="li"
      parentStyles={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    />
    <Placeholder
      childEl="span"
      childStyles={
        Object {
          "fileName": Object {
            "width": "75%",
          },
          "language": Object {
            "backgroundColor": "#5b616b",
            "marginRight": "0",
            "width": "20%",
          },
        }
      }
      key="srt-5680"
      parentEl="li"
      parentStyles={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    />
    <Placeholder
      childEl="span"
      childStyles={
        Object {
          "fileName": Object {
            "width": "75%",
          },
          "language": Object {
            "backgroundColor": "#5b616b",
            "marginRight": "0",
            "width": "20%",
          },
        }
      }
      key="srt-5681"
      parentEl="li"
      parentStyles={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    />
  </ul>
</Fragment>
`;
